src/containers/pages/InterventionPlan/IRS/plan/index.tsx

```diff
@@ -156,14 +156,67 @@ interface IrsPlanState {
   tableCrumbs: TableCrumb[];
 }
 
+let selfIrsPlan: any;
+
 /** IrsPlan - component for IRS Plan page */
 class IrsPlan extends React.Component<
   RouteComponentProps<RouteParams> & IrsPlanProps,
   IrsPlanState
 > {
   public static defaultProps = defaultIrsPlanProps;
+
+  public static getDerivedStateFromProps(
+    props: IrsPlanProps,
+    state: IrsPlanState
+  ): IrsPlanState | null {
+    const {
+      childlessChildrenIds,
+      country,
+      isLoadingGeoms,
+      isLoadingJurisdictions,
+      newPlan,
+    } = state;
+    const { isFinalizedPlan, jurisdictionsById, planById } = props;
+
+    // update state after geometries are fetched from this.loadJurisdictionGeometries()
+    if (newPlan && childlessChildrenIds && country && isLoadingGeoms) {
+      const filteredJurisdictions = childlessChildrenIds.map(j => jurisdictionsById[j]);
+      const loadedJurisdictions = filteredJurisdictions.filter((j: Jurisdiction) => j.geojson);
+
+      if (loadedJurisdictions.length === filteredJurisdictions.length) {
+        const nextState: IrsPlanState = {
+          ...state,
+          isLoadingGeoms: false,
+        };
+        const gisidaWrapperProps = selfIrsPlan.getGisidaWrapperProps(props, nextState);
```

Comment:
It seems that within a static function like this we can't call other methods on the class, whether public or private. I've tried calling `this.getGisidaWrapperProps` and `IrsPlan.getGisidaWrapperProps`, and using an static external variable to reference the class seems a bit hacky.

@moshthepitt - any suggested approaches or do you think this is fine?